client/src/utils/helpers.ts

```diff
@@ -43,10 +43,17 @@ const updateUserField = <T extends keyof IUser>(
  * @param user the user to get the fullname of
  * @returns the fullname of the user
  */
-const getUserFullName = (user: Partial<IUser>): string =>
-  `${user.preferredName ? user.preferredName : user.firstName} ${
+const getUserFullName = (
+  user: IUser | Partial<IUser> | null | undefined,
```

Comment:
I am pretty sure `IUser` is a subset of `Partial<IUser>` so this is unnecessary

Also

`const getUserFullName = (user?: Partial<IUser>) => { ` should be all you need